CPAT.wdl

```diff
@@ -0,0 +1,38 @@
+version 1.0
+
+task CPAT {
+    input {
+        String? preCommand
+        File gene
+        String outFilePath
+        String hex
+        String logitModel
+        File? referenceGenome
+        File? referenceGenomeIndex  # Should be added as input if
+        # CPAT should not index ther reference genome.
+        Array[String]? startCodons
+        Array[String]? stopCodons
+    }
+
+    # Some WDL magic in the command section to properly output the start and stopcodons to the command.
+    command {
+        set -e -o pipefail
+        mkdir -p $(dirname ~{outFilePath})
+        ~{preCommand}
+        cpat.py \
+        --gene ~{gene} \
+        --outfile ~{outFilePath} \
+        --hex ~{hex} \
+        --logitModel ~{logitModel} \
+        ~{"--ref " + referenceGenome} \
+        ~{true="--start" false="" defined(startCodons)} ~{sep="," select_first([startCodons, [""]])} \
```

Comment:
Are you sure `select_first` is needed here?

Reply:
Yes. Optional arrays give problems

Reply:
Also the `""` are needed. As it crashes on an empty array.